WordPress/Classes/Services/BlogService.m

```diff
@@ -159,9 +159,14 @@ - (void)syncBlogsForAccount:(WPAccount *)account
 {
     DDLogMethod();
 
+    NSManagedObjectID *accountID = account.objectID;
     id<AccountServiceRemote> remote = [self remoteForAccount:account];
     [remote getBlogsWithSuccess:^(NSArray *blogs) {
         [self.managedObjectContext performBlock:^{
+            WPAccount *account = (WPAccount *)[self.managedObjectContext existingObjectWithID:accountID error:nil];
+            if (!account) {
+                return;
```

Comment:
@koke I'm beginning to think passing any parameters by `NSManagedObjectID` is going to have to be the requirement. It feels less elegant but it provides a level of safety we don't have. Passing around temporary IDs can also add a level of unexpected behavior. Calling `obtainPermanentIDForObject` is expensive from what I know since it has to go all the way down to the store to obtain the ID. 

We're also not calling refreshObject in many places which feels like a potential source of problems especially if an object is deleted in another context.